src/machine/machine_stm32_flash.go

```diff
@@ -0,0 +1,100 @@
+//go:build stm32f4 || stm32l4 || stm32wlx
+
+package machine
+
+import (
+	"device/stm32"
+
+	"unsafe"
+)
+
+// compile-time check for ensuring we fulfill BlockDevice interface
+var _ BlockDevice = flashBlockDevice{}
+
+var Flash flashBlockDevice
+
+type flashBlockDevice struct {
+}
+
+// ReadAt reads the given number of bytes from the block device.
+func (f flashBlockDevice) ReadAt(p []byte, off int64) (n int, err error) {
+	data := unsafe.Slice((*byte)(unsafe.Pointer(FlashDataStart()+uintptr(off))), len(p))
+	copy(p, data)
+
+	return 0, nil
```

Comment:
Same comment as above ... should return the number of bytes read into the buffer (perhaps additionallly EOF or another error if attempting to read past the end of flash data address)

Reply:
Good point. Both done.